sensible-proxy_test.go

```diff
@@ -22,17 +23,9 @@ func (w *FakeWriter) Write(p []byte) (n int, err error) {
 func TestHTTPConnection(t *testing.T) {
 	w := &FakeWriter{}
 	appLog = log.New(w, "", log.Ldate|log.Ltime)
-	done := make(chan bool)
-	listener, err := getProxyServer(done, handleHTTPConnection)
-	conn, err := net.Dial("tcp", listener.Addr().String())
-	if err != nil {
-		t.Fatal(err)
-	}
 
-	actual, err := requestDomainHTTP(conn, "google.com")
-	if err != nil {
-		t.Errorf("Error on connect: %s", err)
-	}
+	actual, conn, _ := requestHTTP("google.com")
+	defer conn.Close()
```

Comment:
you seem to be doing it for `requestHTTPS`, so maybe make this consistent.
https://github.com/mateusz/sensible-proxy/pull/5/files#diff-2c3c704edfdf526f60eaa169c62db24bR148